pkg/storage/zdb.go

```diff
@@ -98,34 +101,90 @@ func (s *storageModule) Allocate(nsID string, diskType pkg.DeviceType, size uint
 		}
 	}
 
-	type Candidate struct {
-		filesystem.Volume
-		Free uint64
-	}
-
 	targetMode := zdbpool.IndexModeKeyValue
 	if mode == pkg.ZDBModeSeq {
 		targetMode = zdbpool.IndexModeSequential
 	}
 
-	var candidates []Candidate
-	// okay, so this is a new allocation
-	// we search all the pools for a zdb subvolume that can
-	// hold the new namespace
-	for _, pool := range s.volumes {
-		// skip pool with wrong disk type
-		if pool.Type() != diskType {
+	// check for candidates in mounted pools first
+	candidates, err := s.checkForZDBCandidates(size, diskType, true, targetMode)
```

Comment:
is it normal we do not look on the unmounted pool if there are not candidate withint the mounted pool ?

Reply:
Because when it cannot find candidates in mounted pools, then it calls the storagemodule to create a subvol. It's on line 133